File: delivery-java-project/api/src/main/java/org/delivery/api/domain/token/service/TokenService.java
package org.delivery.api.domain.token.service;

import java.util.HashMap;
import java.util.Objects;

import org.delivery.common.error.ErrorCode;
import org.delivery.api.common.exception.ApiException;
import org.delivery.api.domain.token.interfaces.TokenHelperInterface;
import org.delivery.api.domain.token.model.TokenDto;
import org.springframework.stereotype.Service;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Service
public class TokenService {
  private final TokenHelperInterface tokenHelperInterface;

  // 토큰 발급
  public TokenDto issueAccessToken(Long userId) {
    var data = new HashMap<String, Object>();
    data.put("userId", userId);
    return tokenHelperInterface.issueAccessToken(data);
  }

  // 리프레쉬 토큰 발급
  public TokenDto issueRefreshToken(Long userId) {
    var data = new HashMap<String, Object>();
    data.put("userId", userId);
    return tokenHelperInterface.issueRefreshToken(data);
  }

  // 토큰 to User ID
  public Long validationToken(String token) {
    var map = tokenHelperInterface.validationTokenWithThrow(token);
    System.out.println(map.keySet());
    var userId = map.get("userId");

    Objects.requireNonNull(userId, ()-> {
      throw new ApiException(ErrorCode.NULL_POINT);
    });

    return Long.parseLong(userId.toString());
  }
}

File: delivery-java-project/acount/src/main/kotlin/org/delivery/account/domain/token/service/TokenService.kt
package org.delivery.account.domain.token.service

import org.delivery.account.domain.token.ifs.TokenHelperIfs
import org.delivery.account.domain.token.model.TokenDto
import org.springframework.stereotype.Service

@Service
class TokenService(private val tokenHelper: TokenHelperIfs) {
	fun issueAccessToken(userId: Long?): TokenDto? {
		return userId?.let {
			val data = mapOf("userId" to it)
			tokenHelper.issueAccessToken(data)
		}
	}
	
	fun issueRefreshToken(userId: Long?): TokenDto? {
		/// requiredNotNull 구문이 나오고 그 이후는
		/// 그 안에 들어간 데이터가 NotNull임을 의미하고 Null일 경우 에러 발생
		requireNotNull(userId)
		val data = mapOf("userId" to userId)
		return tokenHelper.issueRefreshToken(data)
	}
	
	fun validationTokenWithThrow(token: String?): Long? {
//		 requireNotNull(token)
//		 val map = tokenHelper.validationTokenWithThrow(token)
//		 val userId = map?.get("userId")
//		 requireNotNull(userId)
//		 return userId.toString().toLong()
		
		return tokenHelper
			.validationTokenWithThrow(token)
			?.let { map ->
				map["userId"]
			}.toString().toLong()
	}
}